features/names.feature

```diff
@@ -85,3 +85,35 @@ Feature: BibTeX Names
     | {Barnes} {and} {Noble,} {Inc.}    | {Barnes} {and} {Noble,} |                | {Inc.}                         |     |
     | Charles Louis Xavier Joseph de la Vallee Poussin | Charles Louis Xavier Joseph | de la | Vallee Poussin       |     |
 
+	@names @display
+	Scenarios: Some Japanese names (ordinal names to weird cases)
+		| name                              | first                   | von            | last                           | jr  |
+		| 田中                              |                         |                | 田中                           |     |
+		| 田中 花子                         | 花子                    |                | 田中                           |     |
+		| たなか はなこ                     | はなこ                  |                | たなか                         |     |
+		| タナカ ハナコ                     | ハナコ                  |                | タナカ                         |     |
+		| {山田} 太郎                       | 太郎                    |                | {山田}                         |     |
+		| {山}田 {太郎}                     | {太郎}                  |                | {山}田                         |     |
+		| {アリス} {鈴木}                   | {アリス}                |                | {鈴木}                         |     |
+		| Alice {鈴木}                      | Alice                   |                | {鈴木}                         |     |
+		| 大君 ズム 慈恩                    | ズム 慈恩               |                | 大君                           |     |
+		| {大君 ズム} 慈恩                  | 慈恩                    |                | {大君 ズム}                    |     |
+		| 大君{ }ズム 慈恩                  | 慈恩                    |                | 大君{ }ズム                    |     |
+		| 佐藤 B作                          | B作                     |                | 佐藤                           |     |
+		| 佐藤 b作                          | b作                     |                | 佐藤                           |     |
+		| 佐藤 b作 俊夫                     |                         | 佐藤 b作       | 俊夫                           |     |
+		| 佐藤 {b}作 俊夫                   | {b}作 俊夫              |                | 佐藤                           |     |
+		| 聖in斗 天草 四郎                  | 四郎                    | 聖in斗         | 天草                           |     |
+		| 聖In斗 天草 四郎                  | 天草 四郎               |                | 聖In斗                         |     |
+		| 聖in斗 天草 shi郎                 | shi郎                   | 聖in斗         | 天草                           |     |
+		| Micro soft デベロップメント       |                         | Micro soft     | デベロップメント               |     |
```

Comment:
I just tried to cover all possibilities of parsing. In parsing lower/upper case is distinguished, but these distinction doesn't make sense of Japanese names. OK, I will separate the testcase, one is for ordinal names and the other to cover all posibillities.